transport/uml/auth_flow/should_authorize_diagram.puml

```diff
@@ -0,0 +1,68 @@
+@startuml
+'https://plantuml.com/sequence-diagram
+
+!theme bluegray
+!pragma teoz true
+
+skinparam backgroundColor #whitesmoke
+title shouldAuthorize flow
+
+participant App
+participant MessagingClient
+participant AuthHandler
+participant Vault
+participant WebMessagingAPI
+participant "Genesys Cloud Platform API"
+
+rnote over "App"
+Given: MessagingClient initialized.
+Note: shouldAuthorize might have async operation on the background.
+Therefore (callback: Boolean) can be delayed.
+end rnote
+
+App -> MessagingClient : shouldAuthorize(callback: Boolean)
+MessagingClient -> AuthHandler : shouldAuthorize(callback: Boolean)
+AuthHandler -> AuthHandler : hasRefreshToken()
+alt #aaff8833 hasRefreshToken() == true
+    AuthHandler -> WebMessagingAPI : refreshAuthJwt(refreshToken)
+    WebMessagingAPI -> "Genesys Cloud Platform API" : refreshAuthJwt(refreshToken)
+    alt #aaff8833 Success
+    WebMessagingAPI --> AuthHandler : Result.Success(refreshToken)
+    AuthHandler -> AuthHandler : update refresh and jwt tokens
```

Comment:
My mistake. refreshToken request return only jwt. It does not "update the refresh token". 
So correct statement here should be: "update jwt token"